pkg/backend/endpoint_types.go

```diff
@@ -12,18 +12,18 @@ type Endpoint interface {
 	GetIdentifier(includeTimestamp bool) string
 }
 
-// An endpoint provided by pod directly.
-type PodEndpoint struct {
+// IpEndpoint is an endpoint for an ip address
+type IpEndpoint struct {
 	// Pod's IP.
 	IP string
 	// Pod's container port.
 	Port int32
 	// Pod that provides this endpoint.
-	Pod k8s.PodInfo
+	Pod *k8s.PodInfo
```

Comment:
This needs to remain backwards compatible

Reply:
Why?